domain/src/Commands/DomainCommands.php

```
@@ -0,0 +1,1288 @@
+          // Entity does not support fields.
+          $domain_entities[] = '/' . $type;
+        }
+      }
+      if (array_key_exists(DOMAIN_ACCESS_ALL_FIELD, $fields)) {
+        $domain_affiliate_entities[] = $type;
+      }
+    }
+    $rows['domain_access_entities'] = implode(', ', $domain_entities);
+    $rows['all_affiliate_support'] = implode(', ', $domain_affiliate_entities);
+
+    return new PropertyList($rows);
+  }
+
+  /**
+   * Add a new domain to the site.
+   *
+   * @param $hostname
+   *   The domain hostname to register (e.g. example.com).
+   * @param $name
+   *   The name of the site (e.g. Domain Two).
+   * @param array $options An associative array of options whose values come
+   *   from cli, aliases, config, etc.
+   * @option inactive
+   *   Set the domain to inactive status if set.
+   * @option https
+   *   Use https protocol for this domain if set.
+   * @option weight
+   *   Set the order (weight) of the domain.
+   * @option is_default
+   *   Set this domain as the default domain.
+   * @option validate
+   *   Force a check of the URL response before allowing registration.
+   * @usage drush domain-add example.com 'My Test Site'
+   * @usage drush domain-add example.com 'My Test Site' --inactive=1 --https==1
+   * @usage drush domain-add example.com 'My Test Site' --weight=10
+   * @usage drush domain-add example.com 'My Test Site' --validate=1
+   *
+   * @command domain:add
+   * @aliases domain-add
+   *
+   * @return string
+   *   The entity id of the created domain.
+   *
+   * @throws \Drupal\domain\Commands\DomainCommandException
+   */
+  public function add($hostname, $name, array $options = ['inactive' => null, 'https' => null, 'weight' => null, 'is_default' => null, 'validate' => null]) {
+    $domain_storage = $this->getStorage();
+    $records_count = $domain_storage->getQuery()->count()->execute();
+    $start_weight = $records_count + 1;
+    $hostname = mb_strtolower($hostname);
```

Normally, we throw an error in the validation step rather than silently correcting the lowercase rule.

---

Replaced with a validation step.